nautobot/core/models/__init__.py

```diff
@@ -50,3 +97,92 @@ def validated_save(self):
+        2. Handles variadic natural-keys (e.g. Location model - [name, parent__name, parent__parent__name, ...].)
+        """
+        vals = []
+        for name_list in [name.split("__") for name in self.get_natural_key_fields()]:
+            val = self
+            for attr_name in name_list:
+                val = getattr(val, attr_name)
+                if val is None:
+                    break
+            vals.append(val)
+        # Strip trailing Nones from vals
+        cleaned_vals = []
+        for val in reversed(vals):
+            if val is None and not cleaned_vals:
+                continue
+            cleaned_vals = [val] + cleaned_vals
+        return cleaned_vals
+
+    natural_key_separator = "/"
+
+    @property
+    def natural_key_slug(self):
+        """Less naive implementation than django-natural-keys provides by default, based around URL percent-encoding."""
+        return self.natural_key_separator.join(
+            quote_plus(str(value) if value is not None else "\0") for value in self.natural_key()
+        )
+
+    @classmethod
+    def get_natural_key_def(cls):
+        """
+        Extend django-natural-keys implementation to recognize that UUIDFields are not relevant to the natural key.
+        """
+        if hasattr(cls, "_natural_key"):
+            return cls._natural_key
+
+        for constraint in cls._meta.constraints:
+            if isinstance(constraint, models.UniqueConstraint):
+                return constraint.fields
+
+        if cls._meta.unique_together:
+            return cls._meta.unique_together[0]
+
+        unique = [
+            f
+            for f in cls._meta.fields
+            if f.unique
+            and f.__class__.__name__
+            not in [
+                "AutoField",
+                "BigAutoField",
+                "UUIDField",
+            ]
+        ]
+        if unique:
+            return (unique[0].name,)
+
+        raise Exception(
+            f"Unable to identify an intrinsic natural-key definition for {cls.__name__}. "
+            "If there isn't at least one UniqueConstraint, unique_together, or field with unique=True, "
+            "you probably need to explicitly declare a natural-key for this model."
+        )
+
+    @classmethod
+    def get_natural_key_fields(cls):
+        """
+        Determine actual natural key field list, incorporating the natural keys of related objects as needed.
+
+        Extends django-natural-key's implementation to handle related objects without natural keys (e.g. ContentType).
+        """
+        natural_key_fields = []
+        for name, rel_to in cls.get_natural_key_info():
+            if not rel_to:
+                natural_key_fields.append(name)
+            else:
+                try:
+                    nested_key = rel_to.get_natural_key_fields()
+                except AttributeError:
+                    if rel_to == ContentType:
+                        nested_key = ["app_label", "model"]
```

Comment:
I don't particularly like this special case but it works. If we reimplement/vendor `django-natural-keys` in its entirety, we might try to get rid of `get_natural_key_fields` somehow to where this wouldn't be needed?